staging/src/k8s.io/client-go/util/jsonpath/parser.go

```diff
@@ -332,15 +332,20 @@ Loop:
 
 // parseFilter scans filter inside array selection
 func (p *Parser) parseFilter(cur *ListNode) error {
+	quotes := 0
 	p.pos += len("[?(")
 	p.consumeText()
 Loop:
 	for {
 		switch p.next() {
 		case eof, '\n':
 			return fmt.Errorf("unterminated filter")
+		case '"':
+			quotes = quotes + 1
 		case ')':
-			break Loop
+			if quotes == 0 || quotes == 2 {
```

Comment:
a comment here to the effect that this is ensuring that we close the surrounding quotes would help future reviewers.